homeassistant/components/weather/ecobee.py

```diff
@@ -0,0 +1,146 @@
+        """Return the current condition."""
+        return self.get_forecast(0, 'condition')
+
+    @property
+    def temperature(self):
+        """Return the temperature."""
+        return float(self.get_forecast(0, 'temperature')) / 10
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_FAHRENHEIT
+
+    @property
+    def pressure(self):
+        """Return the pressure."""
+        return int(self.get_forecast(0, 'pressure'))
+
+    @property
+    def humidity(self):
+        """Return the humidity."""
+        return int(self.get_forecast(0, 'relativeHumidity'))
+
+    @property
+    def visibility(self):
+        """Return the visibility."""
+        return int(self.get_forecast(0, 'visibility'))
+
+    @property
+    def wind_speed(self):
+        """Return the wind speed."""
+        return int(self.get_forecast(0, 'windSpeed'))
+
+    @property
+    def wind_bearing(self):
+        """Return the wind direction."""
+        return int(self.get_forecast(0, 'windBearing'))
+
+    @property
+    def attribution(self):
+        """Return the attribution."""
+        station = self.weather['weatherStation']
+        time = self.weather['timestamp']
+        return "Ecobee weather provided by " + station + " at " + time
+
+    @property
+    def forecast(self):
+        """Return the forecast array."""
+        try:
+            forecasts = []
+            for day in self.weather['forecasts']:
+                forecast = {
+                    ATTR_FORECAST_TIME: day['dateTime'],
+                    ATTR_FORECAST_CONDITION: day['condition'],
+                    ATTR_FORECAST_TEMP: float(day['tempHigh']) / 10,
+                }
+                if day['tempHigh'] == MISSING_DATA:
+                    break
+                if day['tempLow'] != MISSING_DATA:
+                    forecast[ATTR_FORECAST_TEMP_LOW] = \
+                        float(day['tempLow']) / 10
+                if day['pressure'] != MISSING_DATA:
+                    forecast[ATTR_FORECAST_PRESSURE] = int(day['pressure'])
+                if day['windSpeed'] != MISSING_DATA:
+                    forecast[ATTR_FORECAST_WIND_SPEED] = int(day['windSpeed'])
+                if day['visibility'] != MISSING_DATA:
+                    forecast[ATTR_FORECAST_WIND_SPEED] = int(day['visibility'])
+                if day['relativeHumidity'] != MISSING_DATA:
+                    forecast[ATTR_FORECAST_HUMIDITY] = \
+                        int(day['relativeHumidity'])
+                forecasts.append(forecast)
+            return forecasts
+        except (ValueError, IndexError):
+            return STATE_UNKNOWN
+
+    def update(self):
+        """Get the latest state of the sensor."""
+        data = ecobee.NETWORK
+        data.update()
+        thermostat = data.ecobee.get_thermostat(self._index)
+        self.weather = thermostat.get('weather', None)
+        logging.error("Weather Update")
```

Comment:
Don't do this. You should add a default logger for this module at the top, then use that. The log level here should be debug, and maybe add `self.weather` to the log.